File: frontend/src/types/snapshot.ts
export interface Snapshot {
  id: number;
  name: string;
  description?: string;
  risk_data: {
    risks: any[];
    snapshot_created_at: string;
    total_risks: number;
  };
  action_items_data?: {
    action_items: any[];
    snapshot_created_at: string;
    total_action_items: number;
  };
  created_at: string;
  created_by: number;
  risk_count: number;
  action_items_count: number;
}

export interface SnapshotCreate {
  name: string;
  description?: string;
}

export interface SnapshotUpdate {
  name?: string;
  description?: string;
}

export interface SnapshotRestore {
  snapshot_id: number;
  confirm: boolean;
}

File: backend/app/schemas/snapshot.py
from datetime import datetime
from typing import Dict, Any, Optional
from pydantic import BaseModel, ConfigDict


class SnapshotBase(BaseModel):
    name: str
    description: Optional[str] = None


class SnapshotCreate(SnapshotBase):
    pass


class SnapshotUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class SnapshotInDB(SnapshotBase):
    id: int
    risk_data: Dict[str, Any]
    action_items_data: Optional[Dict[str, Any]] = None
    created_at: datetime
    created_by: int

    model_config = ConfigDict(from_attributes=True)


class Snapshot(SnapshotInDB):
    risk_count: int
    action_items_count: int


class SnapshotRestore(BaseModel):
    snapshot_id: int
    confirm: bool = False
